SYS_ATL/LoopIR_scheduling.py

```diff
@@ -1799,6 +1799,127 @@ def map_s(self, s):
         return super().map_s(s)
 
 
+# This is same as original FissionAfter, except that
+# this does not remove loop. We have separate remove_loop
+# operator for that purpose.
+class _DoFissionAfterSimple:
+    def __init__(self, proc, stmt, n_lifts):
+        assert isinstance(stmt, LoopIR.stmt)
+        assert is_pos_int(n_lifts)
+        self.orig_proc      = proc
+        self.tgt_stmt       = stmt
+        self.n_lifts        = n_lifts
+
+        self.hit_fission    = False     # signal to map_stmts
+
+        pre_body, post_body = self.map_stmts(proc.body)
+        self.proc = LoopIR.proc(name    = self.orig_proc.name,
+                                args    = self.orig_proc.args,
+                                preds   = self.orig_proc.preds,
+                                body    = pre_body + post_body,
+                                instr   = None,
+                                eff     = self.orig_proc.eff,
+                                srcinfo = self.orig_proc.srcinfo)
+        self.proc = InferEffects(self.proc).result()
+
+    def result(self):
+        return self.proc
+
+    def alloc_check(self, pre, post):
+        if not _is_alloc_free(pre, post):
+            raise SchedulingError("Will not fission here, because "
+                                  "an allocation might be buried "
+                                  "in a different scope than some use-site")
+
+    # returns a pair of stmt-lists
+    # for those statements occurring before and
+    # after the fission point
+    def map_stmts(self, stmts):
+        pre_stmts           = []
+        post_stmts          = []
+        for orig_s in stmts:
+            pre, post       = self.map_s(orig_s)
+            pre_stmts      += pre
+            post_stmts     += post
+
+        return (pre_stmts, post_stmts)
+
+    # see map_stmts comment
+    def map_s(self, s):
+        if s is self.tgt_stmt:
+            #assert self.hit_fission == False
```

Comment:
Why would this assert fail?

Reply:
Because this pass should be executed just once where object s exactly matches with self.tgt_stmt.